src/Dialect/ONNX/ONNXOps/Canonicalize.cpp

```diff
@@ -220,6 +220,25 @@ bool isNegativeSplatConstant(Value val) {
   return false;
 }
 
+/// Test if the input is a constant with all negative value or not.
+bool isAllNegativeIntegerConstant(Value val) {
+  ElementsAttr valAttr = getElementAttributeFromONNXValue(val);
+  if (!valAttr)
+    return false;
+
+  //  if (auto disposable = mlir::dyn_cast<DisposableElementsAttr>(valAttr))
+  //    valAttr = disposable.toDenseElementsAttr();
+
+  Type elemTy = mlir::cast<ShapedType>(val.getType()).getElementType();
+  if (mlir::isa<IntegerType>(elemTy)) {
+    for (auto v : valAttr.getValues<APInt>()) {
+      if (v.getSExtValue() > 0)
+        return false;
+    }
+  }
```

Comment:
It seems we need to return false for the `else` part. Otherwise, it returns true for the float type.

Also, I wonder should we do this check only if the number of values in `val` is small (typically < 10). Otherwise, this check runs for very large constants and it takes time.

Reply:
Done. Thanks!